mlir/test/Dialect/Affine/control-flow-sink.mlir

```diff
@@ -0,0 +1,85 @@
+// RUN: mlir-opt -split-input-file -control-flow-sink %s | FileCheck %s
+
+#set = affine_set<(d0) : (-d0 + 3 >= 0)>
+#map = affine_map<(d0) -> (d0)>
+
+func.func @test_affine_if_sink(%arg1: tensor<4xf32>) -> tensor<4xf32> {
+  %0 = tensor.empty() : tensor<4xf32>
+  %1 = linalg.generic {indexing_maps = [#map, #map], iterator_types = ["parallel"]} ins(%arg1: tensor<4xf32>) outs(%0: tensor<4xf32>) {
+  ^bb0(%in: f32, %out: f32):
+    %2 = linalg.index 0 : index
+    %const0 = arith.constant 0.0 : f32
+    %3 = arith.addf %in, %in: f32
+    %4 = affine.if #set(%2) -> f32 {
+      affine.yield %3 : f32
```

Comment:
```suggestion
    %add = arith.addf %in, %in: f32
    %4 = affine.if #set(%2) -> f32 {
      affine.yield %add : f32
```
Nit: both ops the test cares about would be named in the input when using `add` here